Physical_Engine/Query/InternalElements2D.cs

```diff
@@ -43,6 +43,12 @@ public static partial class Query
         [Output("openings", "The IOpenings of the ISurface.")]
         public static List<IElement2D> InternalElements2D(this oM.Physical.Elements.ISurface surface)
         {
+            if(surface == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the internal 2D elements of a null surface.");
+                return new List<IElement2D>();
```

Comment:
why are we returning an empty list here instead of a `null`? I kind of agree with this taken how core of a method `InternalElements2D` is, but may be worth playing a devil's advocate for a second 👿

Reply:
Mostly to keep consistent with other list returning methods and as changed on the Reflection PR when that engine was done, so we're returning an empty list for consistency with other methods. That and an empty list may be a valid object to work with for whatever was calling it (as you say, this method is quite core to some spatial stuff) and we wouldn't want the other methods to fail on it unnecessarily.

Again, am happy either way.